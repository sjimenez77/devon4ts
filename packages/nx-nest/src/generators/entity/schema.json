{
  "$schema": "https://json-schema.org/schema",
  "$id": "Entity",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "projectName": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 1
      },
      "x-prompt": "In which project would you like to add the entity?"
    },
    "module": {
      "type": "string",
      "description": "Module where the entity will be generated"
    }
  },
  "required": ["name", "projectName"]
}
