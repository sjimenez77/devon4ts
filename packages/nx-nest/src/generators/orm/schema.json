{
  "$schema": "http://json-schema.org/schema",
  "$id": "InitTypeormSchematic",
  "title": "Init Typeorm Schema",
  "type": "object",
  "properties": {
    "orm": {
      "type": "string",
      "enum": ["typeorm"],
      "default": "typeorm"
    },
    "db": {
      "type": "string",
      "enum": ["postgres", "cockroachdb", "mariadb", "mysql", "sqlite", "oracle", "mssql", "mongodb"],
      "default": "sqlite"
    },
    "projectName": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "In which project would you like to add typeorm configuration?"
    }
  },
  "required": ["projectName"]
}
