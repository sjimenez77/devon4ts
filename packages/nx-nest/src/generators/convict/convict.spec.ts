import { ProjectConfiguration, Tree, readProjectConfiguration } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { ApplicationGeneratorOptions } from '@nx/nest/src/generators/application/schema';
import applicationGenerator from '../application/application';
import { convictGenerator } from './convict';

describe('convict generator', () => {
  let tree: Tree;
  const options: ApplicationGeneratorOptions = {
    name: 'test',
    projectNameAndRootFormat: 'as-provided',
    directory: 'apps/test',
    skipFormat: true,
    skipPackageJson: true,
  };
  let appConfig: ProjectConfiguration;
  let loggerConfig: ProjectConfiguration;
  let convictConfig: ProjectConfiguration;

  beforeAll(async () => {
    tree = createTreeWithEmptyWorkspace();
    await applicationGenerator(tree, options);
    await convictGenerator(tree, {
      projectName: options.name,
    });
    await convictGenerator(tree, {
      projectName: options.name,
    });
    appConfig = readProjectConfiguration(tree, options.name);
    loggerConfig = readProjectConfiguration(tree, 'shared-logger');
    convictConfig = readProjectConfiguration(tree, 'shared-config');
  }, 60000);

  it('should run successfully', async () => {
    expect(appConfig).toBeDefined();
    expect(loggerConfig).toBeDefined();
    expect(convictConfig).toBeDefined();
  });

  describe('convict library', () => {
    it('should generate convict library files', () => {
      expect(tree.exists(`${convictConfig.root}/src/lib/base-config.ts`)).toBeTruthy();
      expect(tree.exists(`${convictConfig.root}/src/lib/config.ts`)).toBeTruthy();
      expect(tree.exists(`${convictConfig.root}/src/lib/default-config.ts`)).toBeTruthy();
      expect(tree.exists(`${convictConfig.root}/src/lib/shared-config.module.ts`)).toBeFalsy();
      expect(tree.read(`${convictConfig.root}/src/index.ts`)?.toString('utf-8')).toMatchSnapshot();
    });
  });

  describe('application files', () => {
    it('should add convict configuration to main.ts', async () => {
      expect(tree.read(`${appConfig.root}/src/main.ts`)?.toString('utf-8')).toMatchSnapshot();
    });

    it('should generate config.ts file', async () => {
      expect(tree.read(`${appConfig.root}/src/config.ts`)?.toString('utf-8')).toMatchSnapshot();
    });

    it('should add config provider as global in core.module', () => {
      expect(tree.read(`${appConfig.root}/src/app/core/core.module.ts`)?.toString('utf-8')).toMatchSnapshot();
    });
  });

  describe('root files', () => {
    it('should add dependencies to package.json', async () => {
      const fileContent = tree.read('package.json')?.toString('utf-8');
      expect(fileContent).toMatch(/"dependencies": {(.|\n)*"convict":/g);
      expect(fileContent).toMatch(/"dependencies": {(.|\n)*"@types\/convict":/g);
    });

    it('should update tsconfig.base.json', () => {
      expect(tree.read(`tsconfig.base.json`)?.toString('utf-8')).toMatchSnapshot();
    });
  });

  describe('logger library', () => {
    it('should update logger configuration', async () => {
      expect(tree.read(`${loggerConfig.root}/src/lib/create-base-logger.ts`)?.toString('utf-8')).toMatchSnapshot();
      expect(tree.read(`${loggerConfig.root}/src/lib/logger.module.ts`)?.toString('utf-8')).toMatchSnapshot();
    });
  });
});
