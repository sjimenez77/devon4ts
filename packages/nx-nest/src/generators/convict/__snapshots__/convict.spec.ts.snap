// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convict generator application files should add config provider as global in core.module 1`] = `
"import { ClassSerializerInterceptor, Global, Module } from '@nestjs/common';
import { LoggerModule } from '@proj/shared/logger';
import { APP_INTERCEPTOR } from '@nestjs/core';
import { CONFIG_PROVIDER } from '@proj/shared/config';
import config from '../../config';

@Global()
@Module({
  imports: [LoggerModule],
  controllers: [],
  providers: [
    {
      provide: CONFIG_PROVIDER,
      useValue: config,
    },
    { provide: APP_INTERCEPTOR, useClass: ClassSerializerInterceptor },
  ],
  exports: [CONFIG_PROVIDER, LoggerModule],
})
export class CoreModule {}
"
`;

exports[`convict generator application files should add convict configuration to main.ts 1`] = `
"/**
 * This is not a production server yet!
 * This is only a minimal backend to get started.
 */

import { Logger, ValidationPipe, VersioningType } from '@nestjs/common';
import { NestFactory } from '@nestjs/core';

import { AppModule } from './app/app.module';
import { WinstonLogger } from '@proj/shared/logger';
import { CONFIG_PROVIDER, BaseConfig } from '@proj/shared/config';

async function bootstrap(): Promise<void> {
  const app = await NestFactory.create(AppModule, { bufferLogs: true });
  const config: BaseConfig = app.get(CONFIG_PROVIDER);

  const logger = await app.resolve(WinstonLogger);
  app.useLogger(logger);

  const globalPrefix = config.globalPrefix;
  app.setGlobalPrefix(globalPrefix);
  const port = config.port ?? 3000;
  app.useGlobalPipes(
    new ValidationPipe({
      transform: true,
      transformOptions: {
        excludeExtraneousValues: true,
      },
    }),
  );
  app.enableVersioning({
    type: VersioningType.URI,
    defaultVersion: config.defaultVersion,
  });
  await app.listen(port);
  Logger.log(\`ðŸš€ Application is running on: http://localhost:\${port}/\${globalPrefix}/v\${config.defaultVersion}\`);
}

bootstrap();
"
`;

exports[`convict generator application files should generate config.ts file 1`] = `
"import { BaseConfig, config, defaultConfig } from '@proj/shared/config';
import { Schema } from 'convict';

export default config({
  ...defaultConfig,
} as Schema<BaseConfig>);
"
`;

exports[`convict generator convict library should generate convict library files 1`] = `
"export * from './lib/base-config';
export * from './lib/config';
export * from './lib/default-config';
"
`;

exports[`convict generator logger library should update logger configuration 1`] = `
"/* istanbul ignore file */
import { BaseConfig } from '@proj/shared/config';
import { colorize, oneLineStack, pid } from '@devon4ts/logform';
import * as winston from 'winston';
import { loggerFormatters } from './logger-formatters';

export const BASE_LOGGER = 'BaseLoggerProvider';

export function createBaseLogger(config: BaseConfig): winston.Logger {
  return winston.createLogger({
    format: winston.format.combine(
      winston.format.timestamp(),
      pid(),
      oneLineStack(config.logger.oneLineStack),
      colorize(config.logger.color),
      loggerFormatters[config.logger.formatter](),
    ),
    level: config.logger.loggerLevel,
    transports: new winston.transports.Console(),
  });
}
"
`;

exports[`convict generator logger library should update logger configuration 2`] = `
"import { Module } from '@nestjs/common';
import { BaseConfig, CONFIG_PROVIDER } from '@proj/shared/config';
import winston from 'winston';
import { BASE_LOGGER, createBaseLogger } from './create-base-logger';
import { WinstonLogger } from './winston-logger';

@Module({
  controllers: [],
  providers: [
    WinstonLogger,
    {
      provide: BASE_LOGGER,
      useFactory: (config: BaseConfig): winston.Logger => {
        return createBaseLogger(config);
      },
      inject: [CONFIG_PROVIDER],
    },
  ],
  exports: [WinstonLogger],
})
export class LoggerModule {}
"
`;

exports[`convict generator root files should update tsconfig.base.json 1`] = `
"{
  "compilerOptions": {
    "paths": {
      "@proj/shared/config": ["packages/nx-nest/libs/shared/config/src/index.ts"],
      "@proj/shared/logger": ["packages/nx-nest/libs/shared/logger/src/index.ts"]
    },
    "strictNullChecks": true,
    "noImplicitAny": true,
    "strictBindCallApply": true,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "strict": true,
    "skipDefaultLibCheck": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "skipLibCheck": true,
    "allowSyntheticDefaultImports": true
  }
}
"
`;
